provider/oracle/provider.go

```diff
@@ -0,0 +1,175 @@
+package oracle
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/jsonschema"
+	"github.com/juju/loggo"
+
+	oci "github.com/hoenirvili/go-oracle-cloud/api"
+	"github.com/juju/juju/cloud"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/config"
+)
+
+var logger = loggo.GetLogger("juju.provider.oracle")
+
+const (
+	providerType = "oracle"
+)
+
+// provide friendly aliases for the register provider function
```

Comment:
Aliases exist for backwards compatibility. Please don't add aliases unless they're *needed*. It's not hard for someone to write "oracle", and it reduces the complexity (no need to test that "oracle" and "ocl" and "orcl" and "oci" all mean the same thing.)